src/main/scala/ch/epfl/bluebrain/nexus/rdf/circe/JsonLd.scala

```diff
@@ -3,11 +3,23 @@ package ch.epfl.bluebrain.nexus.rdf.circe
 import ch.epfl.bluebrain.nexus.rdf.Iri
 import ch.epfl.bluebrain.nexus.rdf.Iri.AbsoluteIri
 import ch.epfl.bluebrain.nexus.rdf.jena.JenaModel
-import io.circe.{Json, JsonObject}
 import io.circe.syntax._
+import io.circe.{Json, JsonObject}
 
 object JsonLd {
 
+  /**
+    * Adds @id value to the provided Json
+    *
+    * @param json  the json
+    * @param value the @id value
+    */
+  def id(json: Json, value: AbsoluteIri): Json =
+    json.asObject -> json.asArray match {
+      case (Some(_), _)                      => json deepMerge Json.obj("@id" -> Json.fromString(value.asString))
+      case (_, Some(jArr)) if jArr.size == 1 => Json.arr(id(jArr.head, value))
+    }
```

Comment:
We should treat the default case, at least to throw a more explicit exception than a MatchError.